config/backend-config/single_workspace.go

```diff
@@ -124,21 +97,17 @@ func (wc *singleWorkspaceConfig) getFromAPI(ctx context.Context, _ string) (map[
 	}
 	workspaceID := sourcesJSON.WorkspaceID
 
-	wc.workspaceIDLock.Lock()
-	wc.workspaceID = workspaceID
-	wc.workspaceIDToLibrariesMap = make(map[string]LibrariesT)
-	wc.workspaceIDToLibrariesMap[workspaceID] = sourcesJSON.Libraries
-	wc.workspaceIDLock.Unlock()
+	wc.doOnce.Do(func() {
+		wc.workspaceID = workspaceID
```

Comment:
`workspaceID` will not be initialised properly here in case of a cached configuration. We need to take care of that too.
One way would be to "override" `BackendConfig.Identity` and if we are `usingCache`handle and the `Identifier.ID` is empty we should intervene by providing an augmented `identity.Identifier`.